conan.cmake

```diff
@@ -535,20 +534,23 @@ macro(conan_check)
     endif()
 endmacro()
 
-macro(conan_add_remote)
+function(conan_add_remote)
     # Adds a remote
-    # Arguments URL and NAME are required, INDEX is optional
+    # Arguments URL and NAME are required, INDEX and COMMAND is optional
     # Example usage:
     #    conan_add_remote(NAME bincrafters INDEX 1
     #       URL https://api.bintray.com/conan/bincrafters/public-conan)
-    set(oneValueArgs URL NAME INDEX)
+    set(oneValueArgs URL NAME INDEX COMMAND)
     cmake_parse_arguments(CONAN "" "${oneValueArgs}" "" ${ARGN})
 
     if(DEFINED CONAN_INDEX)
         set(CONAN_INDEX_ARG "-i ${CONAN_INDEX}")
     endif()
-
+    if(CONAN_COMMAND)
```

Comment:
I am not sure I fully understand the flow, and what is being set:
- If ``conan_add_remote()`` defines COMMAND, it will define ``CONAN_CMD`` with its value.
- It will then call ``conan_check()`` macro.
- The ``conan_check()`` calls internally ``find_program(CONAN_CMD conan)``. This will try to find in the path the ``conan`` command, not using the argument provided by the user. Furthermore, it will overwrite the ``CONAN_CMD`` variable value.

It seems the same happens above in ``conan_cmake_install()``. I might be missing some cmake-fu, please tell.

Reply:
There is some magic that here.  In cmake, there is a global variable store, the variable cache.  `find_program(CONAN_CMD conan)` will search first for `CONAN_CMD` on the cache (i.e. in "CMakeCache.txt").  It would be set here with

- set with the `CACHE` flag: `set(CONAN_CMD /path/to/conan CACHE "")`
- cmake from the command line: `cmake -DCONAN_CMD=/path/to/conan ..` (My main motivation for the PR is for this.)
- a previous call to `find_program`

That said, you did highlight a mistake, thank you.  In add remote or cmake install, `CONAN_CMD` is set in the scope of the function and not on the cache.  I put the check in an `else` for the case you highlighted.

Reply:
Great, thanks for the clarification